docs/source/index.rst

```diff
@@ -5,33 +5,35 @@ Tao of workflows
 
-* **Ease of execution of the JSON object should be top priority.** This means where possible complexity should be in the library producing the workflow. As it can be hidden from the user of the library by the tooling within the library.
+* **Ease of execution ASK_MICKY (do we mean processing for execution or the overhead of executing? Because the processing may be complicated but the consumption should be lightweight?) of the JSON object should be top priority.** This means where possible complexity should be in the library producing the workflow. As it can be hidden from the user of the library by the tooling within the library.
 
-* **Minimize surprise.** We should try to make things as least surprising as possible if there is a direct comparison with something in a programing language it should act the same.
+* **Minimize surprise.** We should try to make things the least surprising as possible: if there is a direct comparison with something in a programing language it should act the same.
 
-* **Workflow primitives should be kept to a minimum.** It needs fit in someones head to be understandable. This is another place there tooling is key if you can construct a feature out of primitives do it. Then hide the construction behind tooling. This allows the things required to implement in the client to be minimal.
+* **Workflow primitives should be kept to a minimum.** It needs to fit in a developers head to be understandable and have utility. This is another place there tooling is key; if you can construct a feature from primitives, do. Hide this construction behind tooling. This allows the things requiring implementation in the client to be minimal.
 
-* **Primitives have clearly defined behavior.** It is up to the client to follow those definitions or it breaks the point of being able to drive multiple clients.
+* **Primitives have clearly defined behavior.** It is up to the client to follow those definitions, otherwise it violates the purpose of being implementationally indepedent. ASK_MICKY (does this convey the intention?)
 
-* **Context is transparent looking up the stack but not down.** This means a subflow can see everything in its parent can see but the parent can not look into its subflow. A parent flow only "sees" the result of the subflow. Think of a class function and its return value.
+* **Context is transparent up the stack but not down.** This means a subflow can access everything its parent can, but the parent can not do likewise for a subflow. A parent flow only "sees" the result of the subflow. Think of a class function and its return value.
 
-* **Write tools.** Extending using tools allows for backwards compatibility while also making usage richer.
+* **Write tools.** Extension via tooling allows for backwards compatibility, while also making usage richer. ASK_MICKY (making for a richer usage experience? what do we mean by usage richer?)
```

Comment:
"can do more with less"

Reply:
tools allow for complex composition, without the overhead of direct implementation - you can do more with less